model_compression_toolkit/pytorch/mixed_precision/mixed_precision_wrapper.py

```diff
@@ -43,23 +44,45 @@ def __init__(self,
         self.layer = n.type(**framework_attr)
-        # loading the weights from the graph node (weights of the trained model)
-        self.layer.load_state_dict({k: torch.Tensor(v) for k, v in n.weights.items()}, strict=False)
-        set_model(self.layer)
-
-        self.weight_attrs = fw_info.get_kernel_op_attributes(n.type)
-        # float_weights is a list of weights for each attribute that we want to quantize.
-        self.float_weights = [n.get_weights_by_keys(attr) for attr in
-                              self.weight_attrs]
 
-        assert len(self.weight_attrs) == len(self.float_weights)
-
-        self.node_q_cfg = n.candidates_quantization_cfg
-        self.quantizer_fn_list = [qc.weights_quantization_cfg.weights_quantization_fn for qc in self.node_q_cfg]
-        self.quantized_weights = self._get_quantized_weights()
+        for qc in self.node_q_cfg:
+            assert qc.weights_quantization_cfg.enable_weights_quantization == \
+                   self.node_q_cfg[0].weights_quantization_cfg.enable_weights_quantization \
+                   and qc.activation_quantization_cfg.enable_activation_quantization == \
+                   self.node_q_cfg[0].activation_quantization_cfg.enable_activation_quantization, \
+                "Candidates with different weights/activation enabled properties is currently not supported"
+
+        self.enable_weights_quantization = \
+            self.node_q_cfg[0].weights_quantization_cfg.enable_weights_quantization and \
+            not n.is_all_weights_candidates_equal()
+        self.enable_activation_quantization = \
+            self.node_q_cfg[0].activation_quantization_cfg.enable_activation_quantization and \
+            not n.is_all_activation_candidates_equal()
+
+        # Setting layers' weights
+        if self.enable_weights_quantization:
+            # loading the weights from the graph node (weights of the trained model)
+            self.layer.load_state_dict({k: torch.Tensor(v) for k, v in n.weights.items()}, strict=False)
+            set_model(self.layer)
+
+            self.weight_attrs = fw_info.get_kernel_op_attributes(n.type)
+            # float_weights is a list of weights for each attribute that we want to quantize.
+            self.float_weights = [n.get_weights_by_keys(attr) for attr in
+                                  self.weight_attrs]
+
+            assert len(self.weight_attrs) == len(self.float_weights)
+            self.weights_quantizer_fn_list = [qc.weights_quantization_cfg.weights_quantization_fn
+                                              for qc in self.node_q_cfg]
+            self.quantized_weights = self._get_quantized_weights()
+
+        # Setting layer's activation
+        if self.enable_activation_quantization:
+            self.activation_quantizers = self._get_activation_quantizers()
+            # TODO: How to handle default? (first isn't necessarily the best activation precision)
+            self.activation_bitwidth_idx = 0
```

Comment:
It may be an issue if we configure only part of the layers before we compute their metrics